### ISSUE TITLE
flower api does not work when try to get /api/task/info/(.*)

### ISSUE THREAD

**Issue by lea4n** [mher/flower#708]
when try to access task info, the server reply "HTTP ERROR 500".

my env. is 
1.0.0-dev
4.0.2 (latentcall)
flower   -> flower:1.0.0-dev tornado:4.5.1 babel:2.4.0
software -> celery:4.0.2 (latentcall) kombu:4.0.2 py:2.7.11
            billiard:3.5.0.2 py-amqp:2.1.4
platform -> system:Linux arch:64bit, ELF imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:amqp results:disabled

BTW, how could I get the whole content of args from the flower ui? thanks a lot.

**Comment by johnarnold** [comment]
I don't get that problem, fwiw...

**Comment by lea4n** [comment]
how could I get the whole content of args from the flower ui？
It seems that the ui discard the tail of args, so I can not read the whole content?

**Comment by mher** [comment]
Flower gets task args/kwargs from celery events. If arguments are not small Celery truncates them.

**Comment by mher** [comment]
Please provide stack traces.

**Comment by jheld** [comment]
+1

**Comment by johnarnold** [comment]
https://github.com/mher/flower/pull/719

**Comment by johnarnold** [comment]
FWIW, i did repro this. i'm going to try #719 to fix.

**Comment by johnarnold** [comment]
Tried #719, no fix:

```
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:     HTTPServerRequest(protocol='http', host='nettask.phx.gbl:443', method='GET', uri='/api/task/info/8501f5e1-8a3c-48f5-a3d3-7a15b78c15e4', version='HTTP/1.1', remote_ip='172.17.0.5', headers={'X-Forwarded-Proto': 'https', 'Accept-Language': 'en-US,en;q=0.8', 'Accept': 'application/json', 'X-Real-Ip': '10.20.0.207', 'Accept-Encoding': 'gzip, deflate, br', 'Host': 'nettask.phx.gbl:443', 'Connection': 'upgrade', 'Referer': 'https://nettask.phx.gbl/swagger-ui/', 'X-Forwarded-Port': '443', 'Authorization': 'Basic am9obmFyOjdVKmVxNno1VldYSyNaOQ==', 'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.109 Safari/537.36', 'X-Forwarded-For': '10.20.0.207'})
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:     Traceback (most recent call last):
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/local/lib/python3.5/dist-packages/tornado/web.py", line 1509, in _execute
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         result = method(*self.path_args, **self.path_kwargs)
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/local/lib/python3.5/dist-packages/tornado/web.py", line 2897, in wrapper
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         return method(self, *args, **kwargs)
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/local/lib/python3.5/dist-packages/flower/api/tasks.py", line 604, in get
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         self.write(response)
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/local/lib/python3.5/dist-packages/tornado/web.py", line 715, in write
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         chunk = escape.json_encode(chunk)
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/local/lib/python3.5/dist-packages/tornado/escape.py", line 82, in json_encode
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         return json.dumps(value).replace("</", "<\\/")
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/lib/python3.5/json/__init__.py", line 230, in dumps
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         return _default_encoder.encode(obj)
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/lib/python3.5/json/encoder.py", line 198, in encode
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         chunks = self.iterencode(o, _one_shot=True)
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/lib/python3.5/json/encoder.py", line 256, in iterencode
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         return _iterencode(o, 0)
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:       File "/usr/lib/python3.5/json/encoder.py", line 179, in default
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:         raise TypeError(repr(o) + " is not JSON serializable")
Jul 14 20:25:07 netdocker1-eastus2 INFO a1c9083e3066[1229]:     TypeError: <_weakrefset.WeakSet object at 0x7f4d514b6550> is not JSON serializable
```

**Comment by johnarnold** [comment]
Confirmed fix!

**Comment by mher** [comment]
Fixed by #719
